utils/progressbar/progressbarmng.go

```diff
@@ -189,27 +188,28 @@ func (bm *ProgressBarMng) NewHeadlineBar(msg string) *mpb.Bar {
 func (bm *ProgressBarMng) Increment(prog *TasksWithHeadlineProg) {
 	bm.barsRWMutex.RLock()
 	defer bm.barsRWMutex.RUnlock()
-	if prog.tasksProgressBar.bar.Current() < math.MaxInt64 {
-		prog.tasksProgressBar.bar.Increment()
-	}
+	prog.tasksProgressBar.bar.Increment()
 	prog.tasksProgressBar.tasksCount++
 }
 
 // Increment increments completed tasks count by n.
 func (bm *ProgressBarMng) IncBy(n int, prog *TasksWithHeadlineProg) {
 	bm.barsRWMutex.RLock()
 	defer bm.barsRWMutex.RUnlock()
-	if prog.tasksProgressBar.bar.Current() < math.MaxInt64 {
-		prog.tasksProgressBar.bar.IncrBy(n)
-	}
+	prog.tasksProgressBar.bar.IncrBy(n)
 	prog.tasksProgressBar.tasksCount += int64(n)
 }
 
 // DoneTask increase tasks counter to the number of totalTasks.
 func (bm *ProgressBarMng) DoneTask(prog *TasksWithHeadlineProg) {
 	bm.barsRWMutex.RLock()
 	defer bm.barsRWMutex.RUnlock()
-	prog.tasksProgressBar.bar.SetCurrent(math.MaxInt64)
+	diff := prog.tasksProgressBar.total - prog.tasksProgressBar.tasksCount
+	// Handle large number of total tasks
```

Comment:
I'm not sure this comment is clear enough for the random treader. Let's try explaining clearly the role of the below loop.